src/main/scala/org/broadinstitute/dsde/firecloud/model/ESMapping.scala

```diff
@@ -1,7 +1,81 @@
 package org.broadinstitute.dsde.firecloud.model
 
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.AttributeFormat
+import org.broadinstitute.dsde.firecloud.model.Attributable.AttributeMap
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.json.{JsObject, JsValue}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+
 case class AttributeDefinition(properties: Map[String, AttributeDetail])
 case class AttributeDetail(`type`: String, items: Option[AttributeDetail]=None)
 
 case class ESDatasetProperty(properties: Map[String, ESDetail])
 case class ESDetail(`type`: String)
+
+
+// classes for sending documents to ES to be indexed
+trait Indexable {
+  def id: String
+  def content: JsObject
+}
+
+case class Document(val id: String, val content: JsObject) extends Indexable
+
+object Document {
+  def apply(id: String, valMap: AttributeMap) = {
+    implicit val impAttributeFormat: AttributeFormat = new AttributeFormat with PlainArrayAttributeListSerializer
+    new Document(id, valMap.toJson.asJsObject)
+  }
+  def apply(id: String, jsonStr: String) = new Document(id, jsonStr.parseJson.asJsObject)
+}
+
+
+// classes to convert from json body and to json response
+case class LibrarySearchParams(
+                                searchTerm: Option[String],
```

Comment:
nit: you can leave searchTerm on the previous line and then make sure the from and size to fields line up with it.
(same thing with LibrarySearchResponse below)